mythx_cli/analyze/solidity.py

```diff
@@ -23,6 +23,136 @@ def __init__(self, target: Path):
         self.target = str(target)
         self.payloads = []
 
+    def payload_from_sources(self, solc_result, scribble_file, solc_version):
+        compiled_sources = solc_result.get("sources", {})
+        payload = {
+            "sources": {},
+            "solc_version": solc_version,
+            "main_source": scribble_file or self.target,
+            "source_list": [None] * len(compiled_sources),
```

Comment:
Not sure if this is a common pattern in Python, but have you considered refactoring the payload object keys into variables accessible from a dictionary? For instance
```
common_keys = {
    sources: "sources",
    solc_version: "solc_version",
   etc...
}
```
The benefit of this during development is that you cease to be prone to typing errors, because if you try to do payload[common_keys.sourciz] it'll instantly throw an error saying that the variable `common_keys.sourciz` is undefined.
It also makes it easier to change the value across all the references it has. Just a thought

Reply:
Normally PythX would handle that with an object wrapper. However, in this case we need a `None` list because we need to infer the source list positions from the truffle build artifacts. E.g. we have five artifacts and as a result `[None, None, None, None, None]`. The artifact loop that generates the payload will take the source list ID from each artifact and then fill them into the source list one by one. Sadly, there is no other way to reconstruct the source list as it's not directly part of the artifact data - for whatever reason. :smile: 

This is based on the assumption that truffle artifacts are complete. If that assumption is violated, MythX will also not deliver useful results. Most likely the CLI will catch that in the validation routines as well and cancel the submission early.